src/middlewared/middlewared/utils/account/utmp.py

```diff
@@ -17,13 +19,24 @@
 from middlewared.utils.auth import get_login_uid, AUID_UNSET, AUID_FAULTED
 from middlewared.utils.nss.pwd import getpwuid
 from socket import ntohl
-from threading import Lock  # Generally utmp operations are MT-UNSAFE and race prone
+from threading import RLock  # Generally utmp operations are MT-UNSAFE and race prone
 
 
+__all__ = [
+    'iter_utent', 'utmp_query', 'wtmp_query', 'LoginFile', 'PyUtmpType', 'PyUtmpEntry',
+    'PyUtmpExit', 'login', 'logout',
+]
+
 libc = ctypes.CDLL('libc.so.6', use_errno=True)
-UTMP_LOCK = Lock()
+UTMP_LOCK = RLock()
+MIDDLEWARE_HOST_PREFIX = 'truenas-middleware-session'
```

Comment:
I thought we discussed `tn-mw-session`? I don't mind either way just making a note.

Reply:
I accidentally left that in the file. It's now in `utils/account/authenticator.py` 

```
MIDDLEWARE_HOST_PREFIX = 'tn-mw'
```